apps/analysis/mutation.py

```diff
@@ -269,6 +278,45 @@ class Arguments:
     result = graphene.Field(AnalysisReportUploadType)
 
 
+class CreateAnalysis(RequiredPermissionMixin, PsGrapheneMutation):
+    class Arguments:
+        data = AnalysisInputType(required=True)
+    model = Analysis
+    serializer_class = AnalysisGqlSerializer
+    result = graphene.Field(AnalysisType)
+
+
+class UpdateAnalysis(RequiredPermissionMixin, PsGrapheneMutation):
+    class Arguments:
+        data = AnalysisInputType(required=True)
+        id = graphene.ID(required=True)
+    model = Analysis
+    serializer_class = AnalysisGqlSerializer
+    result = graphene.Field(AnalysisType)
+
+
+class DeleteAnalysis(RequiredPermissionMixin, PsDeleteMutation):
+    class Arguments:
+        id = graphene.ID(required=True)
+    model = Analysis
+    result = graphene.Field(AnalysisType)
```

Comment:
We need to add custom query set to only delete which user have access to.
You can look into existing Delete mutations for ref.